test/test_autograd.py

```diff
@@ -5907,11 +5907,18 @@ def test_saved_variable_packing_unpacking_did_not_save_original_with_default_hoo
             y.sum().backward()
             self.assertEqual(a.grad, y)
 
-    def test_setting_default_saved_variable_hooks_twice_should_fail(self):
-        with self.assertRaisesRegex(RuntimeError, "Setting default hooks but they have already been set. "):
+    def test_setting_default_saved_variable_hooks_twice_should_not_fail(self):
+        with torch.autograd.graph.saved_tensors_hooks(lambda x: x, lambda x: x):
             with torch.autograd.graph.saved_tensors_hooks(lambda x: x, lambda x: x):
-                with torch.autograd.graph.saved_tensors_hooks(lambda x: x, lambda x: x):
-                    pass
+                pass
+
+    def test_setting_default_saved_variable_hooks_twice_should_use_inner(self):
+        with torch.autograd.graph.saved_tensors_hooks(lambda x: 3 * x, lambda x: 3 * x):
+            with torch.autograd.graph.saved_tensors_hooks(lambda x: 5 * x, lambda x: 7 * x):
+                a = torch.randn(5, requires_grad=True)
+                y = a * a
```

Comment:
Could you have some ops in each context to make sure each has the right behavior?